script/tool/test/publish_check_command_test.dart

```diff
@@ -319,6 +319,22 @@ void main() {
           ));
     });
 
+    test('skips packages that are marked as not for publishing', () async {
```

Comment:
I checked  to make sure we had a test for this since I was refactoring the logic that handles this behavior, and we didn't, thus the extra test.